### ISSUE TITLE
Style/EvalWithLocation causing invalid code

### ISSUE THREAD

**Issue by MaxLap** [rubocop/rubocop#11852]
With this code:

```ruby
eval("hello", binding, '123')
```

Running `rubocop -A --only Style/EvalWithLocation the_file.rb`

Generates 
```ruby
eval("hello", binding, __FILE__), __LINE__
```

## Expected behavior

Not sure the cop should run since we are already providing a string for the file... But in any case, I expect it not to produce a SyntaxError.

## Steps to reproduce the problem

Explained in intro

## RuboCop version

```
rubocop -V
1.50.2 (using Parser 3.2.2.1, rubocop-ast 1.28.0, running on ruby 2.6.6) [x86_64-linux]
```
